sdk/core/core-rest-pipeline/review/core-rest-pipeline.api.md

```diff
@@ -199,6 +199,7 @@ export interface PipelineRequest {
     onUploadProgress?: (progress: TransferProgressEvent) => void;
     proxySettings?: ProxySettings;
     requestId: string;
+    responseAsStream?: boolean;
```

Comment:
rather than having to expand surface here, could we reuse `streamResponseStatusCodes` somehow? Like add a special number like  `-1` or Infinity to mean 'always stream'?

Reply:
Yeah we have `streamResponseBody` in core-http but deprecated it. Also what's the scenario where we can not use `streamResponseStatusCodes` with expected status codes?

Reply:
@jeremymeng in Rest Clients, we don't have mappings that would help us mark specific request + statusCode as a streamable response.

In RLCs we have a general client in @azure-rest/core-client, this client allows you to call any method on a resource, for example

```typescript
const result = await getClient("https://example.org").pathUnchecked("/foo").get();
```

When the RLC is generated, for the most part, we just generate types to plug into this general client, no runtime. For example:

```typescript
import { getClient } from "@azure-rest/core-client"

export interface GetConstitution {
  /** The constitution is a script that assesses and applies proposals from consortium members. */
  get(
    options?: GetConstitutionParameters
  ): Promise<GetConstitution200Response | GetConstitutiondefaultResponse>;
}


export interface Routes {
  /** Resource for '/app/governance/constitution' has methods for the following verbs: get */
  (path: "/app/governance/constitution"): GetConstitution;
  /** Resource for '/app/governance/members' has methods for the following verbs: get */
  (path: "/app/governance/members"): GetConsortiumMembers;
}

// This is an example of how a generated RLC looks like. We export this factory method
funtion FooClient() {
   return getClient<Routes >("https://fooservice.org");
}

// Consuming the client from the user perspective

const client = FooClient();
// Only paths defined in Routes are valid when calling "path()". And only the methods defined for that path would show up in intelisense.
const result = await client.path( "/app/governance/constitution").get();
```

So we don't have any runtime information from the swagger, this is by design to keep RLCs simple and away from mappers and declarative logic. So we don't have a way to pass a specific set of status codes for getting a stream response.

The proposed design for RLCs and for what this PR is preparing is to be able to tell RLC when you want a raw stream to handle it yourself.

```typescript
// This gives you a raw stream
const streamResponse = await client.path( "/app/governance/constitution").get().asNodeStream();
handleStream(streamResponse.body)

// This loads the stream in memory and gives you the result, serialized as JSON or string depending on the content-type
const response = await client.path( "/app/governance/constitution").get();
```

Reply:
I also think there's an argument for always getting a stream when a response is successful (i.e. doesn't throw an exception). An API where some status codes return a stream and others a completed body seems less than ideal from the developers standpoint.